tests/unit/Cassandra/DurationTest.php

```diff
@@ -49,6 +49,78 @@ public function testStringArgParseError()
     }
 
     /**
+     * @expectedException InvalidArgumentException
+     * @expectedExceptionMessage nanos must be between -2147483648 and 2147483647, 8589934592 given
+     */
+    public function testStringArgOverflowError()
+    {
+        new Duration(1, 2, "8589934592");
+    }
+
+    /**
+     * @expectedException InvalidArgumentException
+     * @expectedExceptionMessage nanos must be between -2147483648 and 2147483647, -8589934592 given
+     */
+    public function testStringArgUnderflowError()
+    {
+        new Duration(1, 2, "-8589934592");
+    }
+
+    /**
+     * @expectedException InvalidArgumentException
+     * @expectedExceptionMessage nanos must be between -2147483648 and 2147483647, 8589934592 given
+     */
+    public function testBigintArgOverflowError()
+    {
+        new Duration(1, 2, new Bigint("8589934592"));
+    }
+
+    /**
+     * @expectedException InvalidArgumentException
+     * @expectedExceptionMessage nanos must be between -2147483648 and 2147483647, -8589934592 given
+     */
+    public function testBigintArgUnderflowError()
+    {
+        new Duration(1, 2, new Bigint("-8589934592"));
+    }
+
+    /**
+     * @expectedException InvalidArgumentException
+     * @expectedExceptionMessageRegExp /days must be between -2147483648 and 2147483647, 8\.?58993.* given/
```

Comment:
Interesting ... I wonder if it has to do PHP v5.6 using 32-bit longs under the hood and it automatically converts them to a double as the value you are using is larger than PHP_INT_MAX for those platforms.